mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestAccessKey.java

```diff
@@ -0,0 +1,144 @@
+    @Produces(MediaType.APPLICATION_JSON)
+    @MCRRequireAccessKeyAuthorization
+    public Response getAccessKeys(@PathParam(PARAM_MCRID) final MCRObjectID objectId,
+        @DefaultValue("0") @QueryParam("offset") long offset,
+        @DefaultValue("128") @QueryParam("limit") long limit) {
+        if (!MCRMetadataManager.exists(objectId)) {
+            throw getUnknownObjectException();
+        }
+        final long fromIndex = offset;
+        List<MCRAccessKey> accessKeys = MCRAccessKeyManager.getAccessKeys(objectId);
+        final int totalAccessKeyCount = accessKeys.size();
+        if (offset < 0 || limit <= 0 || fromIndex >= Integer.MAX_VALUE || totalAccessKeyCount == 0 
+            || (int) fromIndex >= totalAccessKeyCount) {
+            accessKeys = new ArrayList<MCRAccessKey>();
+        } else {
+            long toIndex = offset + limit;
+            if (toIndex > Integer.MAX_VALUE) {
+                toIndex = Integer.MAX_VALUE;
+            }
+            if ((int) toIndex >= totalAccessKeyCount) {
+                toIndex = (long) totalAccessKeyCount;
+            }
+            accessKeys = accessKeys.subList((int) fromIndex, (int) toIndex);
+        }
+        return Response.ok(new MCRAccessKeyInformation(accessKeys, totalAccessKeyCount)).build();
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @MCRRequireAccessKeyAuthorization
+    @MCRRequireTransaction
+    public Response addAccessKey(@PathParam(PARAM_MCRID) final MCRObjectID objectId, final String accessKeyJson) {
+        final MCRAccessKey accessKey = MCRAccessKeyTransformer.accessKeyFromJson(accessKeyJson);
+        if (!MCRMetadataManager.exists(objectId)) {
+            throw getUnknownObjectException();
+        }
+        accessKey.setObjectId(objectId);
+        accessKey.setCreator(null);
+        accessKey.setCreation(null);
+        accessKey.setLastChanger(null);
+        accessKey.setLastChange(null);
+        MCRAccessKeyManager.addAccessKey(accessKey);
+        return Response.ok(MCRAccessKeyManager.getAccessKeyByValue(objectId, accessKey.getValue())).build();
+    }
+
+    @DELETE
+    @Path("/{" + VALUE + "}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @MCRRequireAccessKeyAuthorization
+    @MCRRequireTransaction
+    public Response deleteAccessKey(@PathParam(PARAM_MCRID) final MCRObjectID objectId, 
+        @PathParam(VALUE) final String value) throws IOException {
+        if (!MCRMetadataManager.exists(objectId)) {
+            throw getUnknownObjectException();
+        }
+        MCRAccessKeyManager.deleteAccessKey(objectId, value);
+        return Response.noContent().build();
+    }
+    
+    @PUT
+    @Path("/{" + VALUE + "}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @MCRRequireAccessKeyAuthorization
+    @MCRRequireTransaction
+    public Response updateAccessKey(@PathParam(PARAM_MCRID) final MCRObjectID objectId, 
```

Comment:
I don't get what this does actually! What information is updated what information is kept here?